File: Assets/UnityChan2D/Demo/FSharpScripts/Loop.fs
namespace UnityChan2D.Demo
open UnityEngine

type Loop () =
  inherit MonoBehaviour()
  [<DefaultValue>] val mutable public to' : Collider2D
  [<DefaultValue>] val mutable public offsetX : float32

  member this.OnTriggerEnter2D(other:Collider2D) =
    if other.tag <> "Player" then () else
    let pos = this.to'.transform.position
    other.transform.position <- new Vector3(pos.x + this.offsetX, other.transform.position.y, other.transform.position.z)

File: Assets/UnityChan2D/Demo/Scripts/Loop.cs
using UnityEngine;
using UniRx;

public class Loop : ObservableMonoBehaviour
{
    public Collider2D to;
    public float offsetX;

    public override void Awake()
    {
        this.OnTriggerEnter2DAsObservable()
            .Where(other => other.tag != "Player")
            .Subscribe(other =>
            {
                var pos = to.transform.position;
                other.transform.position = new Vector2(pos.x + offsetX, other.transform.position.y);
            });

        base.Awake();
    }
}
